src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -673,6 +679,151 @@ public void testOrderOfCheckExecution() throws Exception {
+                        "InputTreeWalkerSkipParsingExceptionConfigSeverityDefault.java");
+        final String[] expected = {
+            "1: " + getCheckMessage(TreeWalker.EXCEPTION_MSG, "IllegalStateException")
+                  + " occurred while parsing file " + path + ".",
+        };
+        verifyWithInlineXmlConfig(path, expected);
+    }
+
+    @Test
+    public void testSkipFileOnJavaParseExceptionSkipChecks() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(TreeWalker.class);
+        config.addProperty("skipFileOnJavaParseException", "true");
+        config.addProperty("javaParseExceptionSeverity", "ignore");
+        config.addChild(createModuleConfig(NoCodeInFileCheck.class));
+
+        final Checker checker = createChecker(config);
+
+        final File[] files = {
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException.java")),
+            new File(getPath("InputTreeWalkerProperFileExtension.java")),
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException2.java")),
+        };
+        final Map<String, List<String>> expectedViolation = new HashMap<>();
+        expectedViolation.put(getPath("InputTreeWalkerProperFileExtension.java"),
+                new ArrayList<>());
+
+        verify(checker, files, expectedViolation);
+    }
+
+    @Test
+    public void testJavaParseExceptionSeverityDefaultError() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(TreeWalker.class);
+        config.addProperty("skipFileOnJavaParseException", "true");
+        config.addChild(createModuleConfig(NoCodeInFileCheck.class));
+
+        final Checker checker = createChecker(config);
+
+        final File[] files = {
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException.java")),
+            new File(getPath("InputTreeWalkerProperFileExtension.java")),
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException2.java")),
+        };
+
+        final Map<String, List<String>> expectedViolation = new HashMap<>();
+
+        expectedViolation.put(getPath("InputTreeWalkerProperFileExtension.java"),
+                new ArrayList<>());
+        expectedViolation.put(getNonCompilablePath("InputTreeWalkerSkipParsingException.java"),
+                List.of("1: Got an exception - IllegalStateException occurred while parsing file "
+                        + getNonCompilablePath("InputTreeWalkerSkipParsingException.java") + "."));
+        expectedViolation.put(getNonCompilablePath("InputTreeWalkerSkipParsingException2.java"),
+                List.of("1: Got an exception - Cannot invoke"
+                        + " \"com.puppycrawl.tools.checkstyle.DetailAstImpl."
+                        + "addPreviousSibling(com.puppycrawl.tools.checkstyle.api.DetailAST)"
+                        + "\" because \"ident\" is null"));
+
+        verify(checker, files, expectedViolation);
+    }
```

Comment:
This test passed locally 

```
[INFO] Running com.puppycrawl.tools.checkstyle.TreeWalkerTest
[INFO] Tests run: 38, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.311 s -- in com.puppycrawl.tools.checkstyle.TreeWalkerTest
```

but fails in CIs 

```
[ERROR] Failures: 
[ERROR]   TreeWalkerTest.testJavaParseExceptionSeverityDefaultError:792->AbstractModuleTestSupport.verify:445->AbstractModuleTestSupport.lambda$verify$1:448 Violations for /home/vsts/work/1/s/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/treewalker/InputTreeWalkerSkipParsingException2.java differ.
value of: iterable.onlyElement()
expected: 1: Got an exception - Cannot invoke "com.puppycrawl.tools.checkstyle.DetailAstImpl.addPreviousSibling(com.puppycrawl.tools.checkstyle.api.DetailAST)" because "ident" is null
but was : 1: Got an exception - null

```

Reply:
> Cannot invoke "com.puppycrawl.tools.checkstyle.DetailAstImpl.addPreviousSibling(com.puppycrawl.tools.checkstyle.api.DetailAST)" because "ident" is null

This message looks JRE specific. Normally NPEs have no message like CI shows.

Reply:
17 passes.
https://github.com/checkstyle/checkstyle/pull/15204/checks?check_run_id=27104642221
11 fails.
https://github.com/checkstyle/checkstyle/pull/15204/checks?check_run_id=27104582489

Reply:
mmm, CI is correct `ex.getMessage()` returns the detailed message not extended message and it is null in the case of the NPE.
I am confused about how this test passed locally it shouldn't 

![1](https://github.com/checkstyle/checkstyle/assets/125591358/497cdd0e-2764-4b9f-a7d5-f40561390481)

Reply:
https://dzone.com/articles/better-default-nullpointerexception-messages-comin
https://bugs.openjdk.org/browse/JDK-8218628

If this is it, it came about in JRE 14+.

Reply:
Yes, it is this. detail message for NPE is null

Reply:
> detail message for NPE is null

On <14.

Reply:
what about introducing a new message something like:
File {file path} is skipped due to a Java parsing exception. so we don't depend on Exception messages.

I like this more. It will be more clear for users that this file is skipped due to parsing exceptions specifically. printing the exact exception message for this case without any different wording is weird to me.

Reply:
> File {file path} 

This should already be part of the violation message.

> that this file is skipped due to 

This may be misleading. The whole file isn't being skipped. It is only not being processed by `TreeWalker`.

> so we don't depend on Exception messages

@nrmancuso What do you think about not showing any part of the exception message? I am not sure if this is a good idea, for the same reason we don't swallow exceptions. Exact message won't mean much to the user, but we will see it in user reports and it could be helpful in debugging.

I am voting for: `TreeWalker is skipped due to parse exception: {Exception}`.

Reply:
IMO, from a user POV lengthy exception messages can be frustrating and annoying. To improve user experience, concise and clear error messages can be more effective. 

If they saw this concise message "TreeWalker is skipped due to parse exception" then they will know that this new syntax in not supported yet and TreeWalker checks are skipped.

Reply:
>  they will know that this new syntax in not supported yet and TreeWalker checks are skipped

I find your faith in the users making this determination disturbing. (star wars)

Reply:
> I am voting for: TreeWalker is skipped due to parse exception: {Exception}

I am good with this.

Reply:
done